packages/DataTable/src/components/ColumnDefinition/ColumnDefinition.js

```diff
@@ -10,12 +10,16 @@ ColumnDefinition.propTypes = {
   momentParsingFormat: PropTypes.string,
   sortDirections: PropTypes.arrayOf(PropTypes.oneOf([sortDirections.ASCEND, sortDirections.DESCEND])),
   type: PropTypes.oneOf([columnTypes.TEXT, columnTypes.NUMBER, columnTypes.DATE]),
+  canHide: PropTypes.bool,
+  isHidden: PropTypes.bool,
 };
 
 ColumnDefinition.defaultProps = {
   momentParsingFormat: null,
   sortDirections: [],
   type: null,
+  canHide: true,
+  isHidden: false,
```

Comment:
shouldn't this be `defaultIsHidden` because you are going to own the state once the consumer passes it? or are you planning to add an effect and watch it? and update the context state?

Reply:
I was planning to add an effect and watch it, so this can be controlled but also uncontrolled